libmscore/scorefile.cpp

```diff
@@ -413,8 +414,36 @@ bool MasterScore::saveFile()
             //
             // step 2
             // remove old backup file if exists
+            // remove the backup file in the same dir as score (the traditional place) if exists
             //
+            QString backupDirString = info.path() + QString(QDir::separator()) + ".backup";
+            if (!dir.cd(".backup")) {
```

Comment:
Is there no better method to detect whether a directory exists? Also 'cd' just tests whether you can change into it (execute permission in UNIX, so Linux and Mac), not whether you can write to it (write permission in UNIX, so Linux and Mac), so the check should be for existence and writability.

Reply:
There is QDir::exists() so better use that.
There is QDir::isReadable(), apparently no QDir::isWritable() though :-(

Reply:
Not sure whether QFileInfo::isWritable() can be used on a directory

Reply:
exists() is used for detecting a file, not a folder. It will check for a file named ".backup" instead of such folder.

Reply:
Not `QDir::exists()`, please try it.

Reply:
Well let me solve this build problem with Visual Studio first ;-) It just updated and everything seems wrong...